NDXINTER/reduce.py

```diff
@@ -1,41 +1,171 @@
+    Get the average angle from logs of motor position
+    :param input_file: The input Nexus file
+    :return: Average (mean) angle from motor position readback
     """
-    print("Running validation")
-    required_files = [input_file]
-    required_dirs = [output_dir]
-    for file_path in required_files:
-        if not os.path.isfile(file_path):
-            raise RuntimeError("Unable to find file: {}".format(file_path))
-    for dir in required_dirs:
-        if not os.path.isdir(dir):
-            raise RuntimeError("Unable to find directory: {}".format(dir))
-    print("Validation successful")
-        
+    run_str = input_file.split("INTER")[1].split(".")[0].strip("0")
+    instrument='INTER'
+    name=instrument+run_str
+    ws=Load(Filename=name, OutputWorkspace='TOF_'+run_str)
+    # Filter the logs for all angles starting from time 0 and use the average of the returned angles
+    (angle_list, average_angle) = FilterLogByTime(ws, 'Theta', StartTime=0)
+    return average_angle, name
 
-def main(input_file, output_dir):
-    validate(input_file, output_dir)
-    
-    # Example of printing some stuff which is captured in autoreduction
-    # output log file 
-    print(web_var)
-    print("input_file = " + str(input_file))
-    print("output_dir = " + str(output_dir))
-    
-    # Copy raw data to output dir.
-    # Note this should only be done if raw files are small and for specific
-    # purpose such as testing 
-    shutil.copy(input_file, output_dir)
-
-    # And of course, here and below insert your reduction code!
+def parse_json_settings(angle):
+    """
+    Get experiment settings and instrument settings from JSON file
+    :param angle: Angle passed in and used to select "per angle defaults"
+    :return: Returns all of the parameters needed to do the reduction
+    """
+    json_input = r"C:\Users\wyf59278\Documents\repos\reflectometry_reduction\settings.json"
+
+    with open(json_input, "r") as read_file:
+        data = json.load(read_file)
+
+    #========================================================================================
+    # Experiment Settings
+    #========================================================================================
+
+    experimentView = data["experimentView"]
+
+    # Set a string based on what integer value is found
+    if experimentView["analysisModeComboBox"] == 1:
+        analysis_mode = "MultiDetectorAnalysis"
+    elif experimentView["analysisModeComboBox"] == 0:
+        analysis_mode = "PointDetectorAnalysis"
+    else:
+        raise Exception # If the value isn't 1 or 0 then it isn't valid
+
+    perAngleDefaults = experimentView["perAngleDefaults"]
+    rows = perAngleDefaults["rows"]
+
+    # This looks for the run angle and set other parameters accordingly
+    # Using a tolerance of +-0.5% of the motor readback angle
+    min = angle * 0.995
+    max = angle * 1.005
+    angle_found = False
+    for i in rows:

```

Comment:
I think `row` is a better name than `i`, you'll have to rename it in the scope below though